otelzap/core.go

```diff
@@ -33,7 +33,8 @@ const (
 type otlpCore struct {
 	logger otel.Logger
 
-	fields []zapcore.Field
+	fields   []zapcore.Field
+	minLevel zapcore.Level
```

Comment:
Can we implement this with `zapcore.LevelEnabler`, I think it would be more align with the zap's implementation of its core 

e.g. 

```go
type otlpCore struct {
    zapcore.LevelEnabler
	logger   otel.Logger
	fields   []zapcore.Field
}
```

and thus we no longer need to re-implement custom `Enabled` function

Reply:
Good idea. I'll look at this

Reply:
Looking at `zapcore.LevelEnabler` I see it's an interface, were you think of a specific implementation of it already present in zap?

Reply:
so the `zapcore.InfoLevel (zapcore.Level)` that you're using is already is an implementing `LevelEnabler` interface and thus we can just directly use it here as well

Reply:
Just pushed a commit that make the current Enabled method rely on the underlying level implementation. Let me know if it still needs more changes.